datalad_registry/tasks.py

```diff
@@ -70,6 +97,37 @@ def _extract_annex_info(repo) -> InfoType:
     return info
 
 
+@celery.task
+def collect_dataset_uuid(urls: Optional[List[str]] = None) -> None:
+    from flask import current_app
+
+    cache_dir = Path(current_app.config["DATALAD_REGISTRY_DATASET_CACHE"])
+    cache_dir.mkdir(parents=True, exist_ok=True)
+
+    if not urls:
+        lgr.debug("Did not find URLs that needed UUIDs collected")
+        return
+
+    lgr.info("Collecting UUIDs for %s URLs", len(urls))
+    lgr.debug("URLs: %s", urls)
+
+    for url in urls:
+        result = db.session.query(URL).filter_by(url=url)
+        if result.first() is not None:
```

Comment:
I know that this is how it is used elsewhere, but I wondered if there is some `.only` or `.first_only` which would otherwise error out if query returns multiple? (which shouldn't happen, so it is just to ensure to "error out than to chase the ghosts")

`url = db.Column(db.Text, primary_key=True)` and I do not see any other key, so I guess it ensures that only one record should be there.

Reply:
That would be the [`scalar_one()`](https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.Result.scalar_one method.

Reply:
let's start using it then wherever it or `scalar_one_or_none` in this case I guess

Reply:
Done.

Reply:
It turns out that `scalar_one()` and `scalar_one_or_none()` are methods of the `Result` type but not of the `Query` type used by the ORM.  Hence, we need to stick with `first()`.